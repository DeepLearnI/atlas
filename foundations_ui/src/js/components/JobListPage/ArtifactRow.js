import React from 'react';
import PropTypes from 'prop-types';

class ArtifactRow extends React.Component {
  constructor(props) {
    super(props);

    this.onClickDownload = this.onClickDownload.bind(this);
    this.onClickThisArtifact = this.onClickThisArtifact.bind(this);
  }

  onClickDownload() {
    const { artifact } = this.props;
    window.open(artifact.uri);
  }

  onClickThisArtifact() {
    const { onClickArtifact, artifact } = this.props;
    onClickArtifact(artifact);
  }

  render() {
    const { artifact } = this.props;

    return (
      <div
        tabIndex="0"
        onKeyPress={this.onClickThisArtifact}
        role="button"
        onClick={this.onClickThisArtifact}
        className="table-artifacts-row"
        data-class="artifacts-table-row"
      >
        <div className="table-artifacts-row-cell">
          <p>{artifact.filename}</p>
        </div>
        <div className="table-artifacts-row-cell last">
          <div className="table-artifacts-row">
            <div
              className="button-download"
              onClick={this.onClickDownload}
              role="button"
              aria-label="Close"
              onKeyDown={this.onKeyDown}
              tabIndex={0}
            >
              <div className="i--icon-download" />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

ArtifactRow.propTypes = {
  artifact: PropTypes.object,
  onClickArtifact: PropTypes.func,
};

ArtifactRow.defaultProps = {
  artifact: {},
  onClickArtifact: () => {},
};

export default ArtifactRow;
