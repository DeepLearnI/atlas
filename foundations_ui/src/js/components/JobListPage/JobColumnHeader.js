import React, { Component } from 'react';
import PropTypes from 'prop-types';


class JobColumnHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: this.props.title,
      isStatus: this.props.isStatus,
    };
  }

  render() {
    const { title, isStatus } = this.state;
    let headerClassName = 'header-4 blue-border-bottom';
    let arrowClassName = 'arrow-down float-right';
    if (isStatus === 1) {
      headerClassName = 'header-4 blue-border-bottom status-header';
      arrowClassName = 'arrow-down margin-auto';
    }

    return (
      <div className="job-column-header">
        <h4 className={headerClassName}>{title}</h4>
        <div className={arrowClassName} />
      </div>
    );
  }
}

JobColumnHeader.propTypes = {
  title: PropTypes.string,
  isStatus: PropTypes.number,
};

JobColumnHeader.defaultProps = {
  title: '',
  isStatus: 0,
};

export default JobColumnHeader;
