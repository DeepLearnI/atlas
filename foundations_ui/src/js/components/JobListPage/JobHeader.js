import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ShowMoreFilters from '../common/filters/ShowMoreFilters';
import CommonActions from '../../actions/CommonActions';

const borderSize = 3; // 1px per side + space between bubbles

class JobHeader extends Component {
  constructor(props) {
    super(props);
    this.toggleFilters = this.toggleFilters.bind(this);
    this.clickRemoveFilter = this.clickRemoveFilter.bind(this);
    this.showHideBubbles = this.showHideBubbles.bind(this);
    this.addIfNotHidden = this.addIfNotHidden.bind(this);
    this.removeFromHidden = this.removeFromHidden.bind(this);
    this.getHiddenWidth = this.getHiddenWidth.bind(this);
    this.addToBubbleRefs = this.addToBubbleRefs.bind(this);
    this.removeBubbleFromRef = this.removeBubbleFromRef.bind(this);
    this.modifyBubble = this.modifyBubble.bind(this);
    this.refsInFilters = this.refsInFilters.bind(this);
    this.getCurHiddenBubbles = this.getCurHiddenBubbles.bind(this);
    this.removeHiddenButtonCallback = this.removeHiddenButtonCallback.bind(this);
    this.state = {
      project: this.props.project,
      filters: this.props.filters,
      bubbleRefs: [],
      isShowingMoreFilters: false,
      clearFilters: this.props.clearFilters,
      removeFilter: this.props.removeFilter,
      hiddenBubbles: [],
    };
  }

  async componentWillReceiveProps(nextProps) {
    this.setState({ filters: nextProps.filters, project: nextProps.project });

    const { hiddenBubbles } = this.state;
    const newRefs = this.refsInFilters(nextProps.filters);

    let curWidth = 0;
    let curHiddenBubbles = CommonActions.deepCopyArray(hiddenBubbles);
    curHiddenBubbles = this.getCurHiddenBubbles(newRefs, curHiddenBubbles);
    newRefs.sort((a, b) => { return a.length - b.length; });
    // newRefs.forEach((id) => {
    //   const showHideResults = this.showHideBubbles(id, curWidth, clientWidth, curHiddenBubbles);
    //   curWidth = showHideResults.width;
    //   curHiddenBubbles = showHideResults.hiddenBubbles;
    // });
    // await this.setState({ bubbleRefs: [], hiddenBubbles: curHiddenBubbles });
  }

  refsInFilters(filters) {
    return filters.map((filter) => {
      return `${filter.column}-${filter.value}`;
    });
  }

  getCurHiddenBubbles(newRefs, hiddenBubbles) {
    return hiddenBubbles.filter((bubble) => {
      return newRefs.indexOf(bubble) >= 0;
    });
  }

  showHideBubbles(id, curWidth, clientWidth, hiddenBubbles) {
    let bubbleWidth = curWidth;
    let curHiddenBubbles = CommonActions.deepCopyArray(hiddenBubbles);
    if (id !== null) {
      const bubble = document.getElementById(id);
      if (bubble !== null) {
        const modifyBubbleResults = this.modifyBubble(
          curHiddenBubbles, id, bubble, bubbleWidth, clientWidth,
        );
        bubbleWidth = modifyBubbleResults.width;
        curHiddenBubbles = modifyBubbleResults.hiddenBubbles;
      }
    }
    return { width: bubbleWidth, hiddenBubbles: curHiddenBubbles };
  }

  modifyBubble(hiddenBubbles, id, bubble, curWidth, clientWidth) {
    let bubbleWidth = curWidth;
    let curHiddenBubbles = CommonActions.deepCopyArray(hiddenBubbles);
    const hiddenWidth = this.getHiddenWidth(hiddenBubbles, id);
    const curBubbleWidth = CommonActions.addBorderToElementWidth(bubble, borderSize, hiddenWidth);
    bubbleWidth += curBubbleWidth;
    if (CommonActions.elementsWidthLargerThanParent(bubbleWidth, clientWidth)) {
      if (bubble && bubble.className && !bubble.className.includes(' hidden')) {
        bubble.className += ' hidden';
        curHiddenBubbles = this.addIfNotHidden(curHiddenBubbles, id, curBubbleWidth);
      }
    } else if (bubble && bubble.className && bubble.className.includes(' hidden')) {
      bubble.className = bubble.className.replace(' hidden', '');
      curHiddenBubbles = this.removeFromHidden(curHiddenBubbles, id);
    }
    return { width: bubbleWidth, hiddenBubbles: curHiddenBubbles };
  }

  addIfNotHidden(array, id, width) {
    const newHidden = this.removeFromHidden(array, id);
    newHidden.push({ id, width });
    return newHidden;
  }

  removeFromHidden(array, id) {
    let newHidden = CommonActions.deepCopyArray(array);
    newHidden = newHidden.filter(
      (filter) => {
        return (filter.id && filter.id !== id) || (!filter.id && filter !== id);
      },
    );
    return newHidden;
  }

  getHiddenWidth(array, id) {
    let newHidden = CommonActions.deepCopyArray(array);
    newHidden = newHidden.filter(
      (filter) => {
        return (filter.id === id);
      },
    );
    if (newHidden[0]) {
      return newHidden[0].width;
    }
    return null;
  }

  toggleFilters() {
    const { isShowingMoreFilters, hiddenBubbles } = this.state;
    if (hiddenBubbles.length > 0) {
      this.setState({ isShowingMoreFilters: !isShowingMoreFilters });
    }
  }

  async clickRemoveFilter(filter) {
    const { removeFilter, bubbleRefs } = this.state;
    const id = filter.column.concat('-').concat(filter.value);
    const currentBubbleRefs = this.removeBubbleFromRef(bubbleRefs, id);
    await removeFilter(filter);
    await this.setState({ bubbleRefs: currentBubbleRefs });
  }

  addToBubbleRefs(key) {
    const { bubbleRefs } = this.state;
    if (!bubbleRefs.includes(key)) {
      bubbleRefs.push(key);
    }
  }

  removeBubbleFromRef(bubbleRefs, id) {
    return bubbleRefs.filter((bubble) => {
      if (bubble !== id) {
        return true;
      }
    });
  }

  removeHiddenButtonCallback(filterToRemove) {
    this.clickRemoveFilter(filterToRemove);
  }

  render() {
    const {
      project, filters, isShowingMoreFilters, clearFilters, hiddenBubbles,
    } = this.state;

    const filterBubbles = [];
    filters.forEach((filter) => {
      const key = filter.column.concat('-').concat(filter.value);
      filterBubbles.push(
        <div ref={() => { this.addToBubbleRefs(key); }} id={key} key={key} className="bubble inline-block">
          <p className="font-bold">
            {filter.column}:<span> {filter.value}</span>
          </p>
          <button onClick={() => { this.clickRemoveFilter(filter); }} type="button" className="close-button" />
        </div>,
      );
    });

    let moreBubbles = null;
    if (hiddenBubbles.length > 0) {
      moreBubbles = (
        <div className="bubble more-bubble">
          <p className="font-bold text-blue text-upper">
          + {hiddenBubbles.length} More
          </p>
        </div>
      );
    }

    let moreFilters = null;
    let filterButtonText = 'View Filters';
    if (isShowingMoreFilters) {
      moreFilters = (
        <ShowMoreFilters
          hiddenBubbles={hiddenBubbles}
          removeFilterCallback={this.removeHiddenButtonCallback}
        />
      );
      filterButtonText = 'Hide Filters';
    }
    let viewFilterClass = 'b--mat b--affirmative text-upper';
    if (hiddenBubbles.length === 0) {
      viewFilterClass += ' b--disabled';
      moreFilters = null;
    }
    let clearFiltersClass = 'b--mat b--affirmative text-upper';
    if (filters.length === 0) {
      clearFiltersClass += ' b--disabled';
    }

    return (
      <div className="job-header-container">
        <div className="job-header-container-left">

          <div className="job-summary-info-container">
            <h2 className="font-bold project-list-name">{project.name}</h2>
            <h3>Jobs List</h3>
            <ul>
              <li>
                <p className="font-bold">Data Source: </p>
                <p>raw_mnist</p>
              </li>
              <li>
                <p className="font-bold">Project owner: </p>
                <p>Trial<span>{project.owner}</span></p>
              </li>
              <li>
                <p className="font-bold">Created at: </p>
                <p>2019-03-30 T 10:45 UTC<span>{project.created_at}</span></p>
              </li>
            </ul>
          </div>
          <div className="job-header-sorting-container">
            {/* <button
              type="button"
              onClick={clearFilters}
              className={clearFiltersClass}
            >
              Clear Filters
            </button>
            <div>
              <div ref={(e) => { this.bubbleContainer = e; }}>
                {filterBubbles}
              </div>
              {moreBubbles}
            </div>
            <div>
              <button
                type="button"
                onClick={this.toggleFilters}
                className={viewFilterClass}
              >
                {filterButtonText}
              </button>
            </div> */}
          </div>
          {moreFilters}
        </div>
      </div>
    );
  }
}

JobHeader.propTypes = {
  numProjects: PropTypes.number,
  project: PropTypes.object,
  filters: PropTypes.array,
  bubbleRefs: PropTypes.array,
  isShowingMoreFilters: PropTypes.bool,
  clearFilters: PropTypes.func,
  removeFilter: PropTypes.func,
  hiddenBubbles: PropTypes.array,

};

JobHeader.defaultProps = {
  numProjects: 0,
  project: { owner: 'null', created_at: 'null', name: 'null' },
  filters: [],
  bubbleRefs: [],
  isShowingMoreFilters: false,
  clearFilters: () => {},
  removeFilter: () => {},
  hiddenBubbles: [],
};

export default JobHeader;
