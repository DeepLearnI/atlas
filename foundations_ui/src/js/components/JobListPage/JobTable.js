import React, { Component } from 'react';
import PropTypes from 'prop-types';
import JobTableHeader from './JobTableHeader';
import JobTableButtons from './JobTableButtons';
import rowSelect from '../../../scss/jquery/rowSelect';


class JobTable extends Component {
  constructor(props) {
    super(props);
    this.onDataUpdated = props.onDataUpdated.bind(this);
    this.onClickOpenModalJobDetails = this.onClickOpenModalJobDetails.bind(this);
    this.updateFilterSearchText = this.updateFilterSearchText.bind(this);
    this.filterColumns = this.filterColumns.bind(this);
    this.updateHiddenColumns = this.updateHiddenColumns.bind(this);
    this.sortTable = this.sortTable.bind(this);
    this.selectJob = this.selectJob.bind(this);
    this.selectAllJobs = this.selectAllJobs.bind(this);
    this.selectNoJobs = this.selectNoJobs.bind(this);
    this.state = {
      jobs: this.props.jobs,
      isLoaded: this.props.isLoaded,
      allInputParams: this.props.allInputParams,
      allMetrics: this.props.allMetrics,
      statuses: this.props.statuses,
      updateHiddenStatus: this.props.updateHiddenStatus,
      updateHiddenUser: this.props.updateHiddenUser,
      updateNumberFilter: this.props.updateNumberFilter,
      updateContainsFilter: this.props.updateContainsFilter,
      updateBoolFilter: this.props.updateBoolFilter,
      updateDurationFilter: this.props.updateDurationFilter,
      updateJobIdFilter: this.props.updateJobIdFilter,
      updateStartTimeFilter: this.props.updateStartTimeFilter,
      allUsers: this.props.allUsers,
      hiddenUsers: this.props.hiddenUsers,
      numberFilters: this.props.numberFilters,
      boolFilters: this.props.boolFilters,
      boolCheckboxes: this.props.boolCheckboxes,
      durationFilters: this.props.durationFilters,
      jobIdFilters: this.props.jobIdFilters,
      startTimeFilters: this.props.startTimeFilters,
      filters: this.props.filters,
      filterSearchText: '',
      filteredColumns: null,
      hiddenColumns: [],
      sortedColumn: { column: '', isAscending: true },
      selectedJobs: [],
      allJobsSelected: false,
      projectName: this.props.projectName,
      getJobs: this.props.getJobs,
      reload: this.props.reload,
      tensorboardEnabled: false,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState(
      {
        statuses: nextProps.statuses,
        jobs: nextProps.jobs,
        isLoaded: nextProps.isLoaded,
        allInputParams: nextProps.allInputParams,
        allMetrics: nextProps.allMetrics,
        allUsers: nextProps.allUsers,
        hiddenUsers: nextProps.hiddenUsers,
        boolCheckboxes: nextProps.boolCheckboxes,
        numberFilters: nextProps.numberFilters,
        boolFilters: nextProps.boolFilters,
        durationFilters: nextProps.durationFilters,
        jobIdFilters: nextProps.jobIdFilters,
        startTimeFilters: nextProps.startTimeFilters,
        filters: nextProps.filters,
        projectName: nextProps.projectName,
      },
    );
  }

  handleRowSelection(rowNumber) {
    const { selectedRow } = this.state;
    if (selectedRow === rowNumber) {
      this.setState({ selectedRow: -1 });
      rowSelect.deselect(rowNumber);
    } else {
      rowSelect.select(rowNumber);
      this.setState({ selectedRow: rowNumber });
    }
  }

  closeSideBar() {
    this.setState({ selectedRow: -1 });
    rowSelect.removePreviousActiveRows();
  }

  onClickOpenModalJobDetails(job) {
    const { onClickJob } = this.props;
    onClickJob(job);
    this.handleRowSelection(job.job_id);
  }

  async updateFilterSearchText(newText) {
    await this.setState({ filterSearchText: newText });
    this.filterColumns();
  }

  filterColumns() {
    const { filterSearchText, allMetrics, allInputParams } = this.state;
    const allFilterableColumns = allMetrics.concat(allInputParams);
    if (filterSearchText.trim().length > 0) {
      const filteredColumns = allFilterableColumns.filter(col => {
        return col.name.includes(filterSearchText);
      });
      this.setState({ filteredColumns: filteredColumns });
    } else {
      this.setState({ filteredColumns: allFilterableColumns });
    }
  }

  updateHiddenColumns(newHidden) {
    this.setState({ hiddenColumns: newHidden });
  }

  sortTable(clickedColumn, mainHeader) {
    const { sortedColumn, getJobs } = this.state;

    let clickedColumnWithMainHeader = clickedColumn;
    let isAscending = true;

    if (mainHeader === 'Parameters') {
      clickedColumnWithMainHeader = `parameters:${clickedColumn}`;
    }
    if (mainHeader === 'Metrics') {
      clickedColumnWithMainHeader = `output_metrics:${clickedColumn}`;
    }

    if (clickedColumnWithMainHeader === sortedColumn.column) {
      if (sortedColumn.isAscending === true) {
        clickedColumnWithMainHeader = '';
      }
    } else {
      isAscending = false;
    }

    this.setState({ sortedColumn: { column: clickedColumnWithMainHeader, isAscending: isAscending } }, () => {
      getJobs(this.state.sortedColumn);
    });
  }

  selectJob(jobID) {
    const { selectedJobs, jobs } = this.state;
    let newSelectedJobs = Array.from(selectedJobs);
    if (selectedJobs.includes(jobID)) {
      newSelectedJobs = newSelectedJobs.filter(job => {
        return job !== jobID;
      });
    } else {
      newSelectedJobs.push(jobID);
      const selectedJob = jobs.find(job => {
        return job.job_id === jobID;
      });
    }
    let newAllJobsSelected = false;
    if (jobs.length === newSelectedJobs.length) {
      newAllJobsSelected = true;
    }

    const newTensorboardEnabled = this.canGoToTensorboard(newSelectedJobs);

    this.setState({
      selectedJobs: newSelectedJobs,
      allJobsSelected: newAllJobsSelected,
      tensorboardEnabled: newTensorboardEnabled,
    });
  }

  canGoToTensorboard(newSelectedJobs) {
    const { jobs } = this.state;
    const jobsByID = {};
    jobs.forEach(job => {
      jobsByID[job.job_id] = job;
    });
    const selectedJobObjects = newSelectedJobs.map(singleJobID => {
      return jobsByID[singleJobID];
    });
    return !!selectedJobObjects.length && selectedJobObjects.every(job => {
      return !!job.tags.tf;
    });
  }

  selectAllJobs() {
    const { jobs, selectedJobs, allJobsSelected } = this.state;
    let jobIds = [];
    let areAllSelected = false;
    if (selectedJobs.length !== jobs.length) {
      jobIds = jobs.map(job => {
        return job.job_id;
      });
      areAllSelected = true;
    }
    const newTensorboardEnabled = this.canGoToTensorboard(jobIds);
    this.setState({
      selectedJobs: jobIds,
      allJobsSelected: areAllSelected,
      tensorboardEnabled: newTensorboardEnabled,
    });
  }

  selectNoJobs() {
    this.setState({ selectedJobs: [], allJobsSelected: false });
  }

  render() {
    const {
      jobs, isLoaded, allInputParams, allMetrics, statuses, updateHiddenStatus, updateHiddenUser, allUsers, hiddenUsers,
      updateNumberFilter, numberFilters, updateContainsFilter, cotainFilters, updateBoolFilter, boolFilters,
      boolCheckboxes, updateDurationFilter, durationFilters, updateJobIdFilter, jobIdFilters, updateStartTimeFilter,
      startTimeFilters, filters, filteredColumns, hiddenColumns, sortedColumn, selectedJobs, allJobsSelected,
      projectName, getJobs, reload, tensorboardEnabled,
    } = this.state;

    const jobRows = [];
    const rowNum = 1;
    const rowNumbers = [];

    const handleClick = job => {};

    const allFilterableColumns = allMetrics.concat(allInputParams);
    const visibleMetrics = allMetrics.filter(col => {
      return !hiddenColumns.includes(col.name);
    });
    const visibleParams = allInputParams.filter(col => {
      return !hiddenColumns.includes(col.name);
    });

    const curVisibleColumns = filteredColumns !== null && filteredColumns.length < allFilterableColumns.length
      ? filteredColumns : allFilterableColumns;

    curVisibleColumns.forEach(col => {
      if (hiddenColumns.includes(col.name)) {
        col.hidden = true;
      } else {
        col.hidden = false;
      }
    });

    return (
      <div className="job-table-content">
        <div className="job-table-container">
          <JobTableButtons
            columns={curVisibleColumns}
            updateSearchText={this.updateFilterSearchText}
            hiddenColumns={hiddenColumns}
            updateHiddenColumns={this.updateHiddenColumns}
            selectAllJobs={this.selectAllJobs}
            selectedJobs={selectedJobs}
            projectName={projectName}
            getJobs={getJobs}
            selectNoJobs={this.selectNoJobs}
            reload={reload}
            buttonTensorboardEnabled={tensorboardEnabled}
          />
          <JobTableHeader
            allInputParams={visibleParams}
            allMetrics={visibleMetrics}
            jobs={jobs}
            statuses={statuses}
            updateHiddenStatus={updateHiddenStatus}
            rowNumbers={rowNumbers}
            jobRows={jobRows}
            updateHiddenUser={updateHiddenUser}
            allUsers={allUsers}
            hiddenUsers={hiddenUsers}
            boolCheckboxes={boolCheckboxes}
            updateNumberFilter={updateNumberFilter}
            numberFilters={numberFilters}
            updateContainsFilter={updateContainsFilter}
            updateBoolFilter={updateBoolFilter}
            boolFilters={boolFilters}
            updateDurationFilter={updateDurationFilter}
            durationFilters={durationFilters}
            updateJobIdFilter={updateJobIdFilter}
            jobIdFilters={jobIdFilters}
            updateStartTimeFilter={updateStartTimeFilter}
            startTimeFilters={startTimeFilters}
            filters={filters}
            onMetricRowClick={handleClick}
            onDataUpdated={this.onDataUpdated}
            onClickOpenModalJobDetails={this.onClickOpenModalJobDetails}
            sortedColumn={sortedColumn}
            sortTable={this.sortTable}
            selectJob={this.selectJob}
            selectAllJobs={this.selectAllJobs}
            selectedJobs={selectedJobs}
            allJobsSelected={allJobsSelected}
          />
          {/* <div className="pagination-controls">
            <p><span className="font-bold">Viewing:</span> 1-100/600</p>
            <div className="arrow-right" />
            <p>Page 1</p>
            <div className="arrow-left" />
          </div> */}
        </div>
      </div>
    );
  }
}

JobTable.propTypes = {
  isMount: PropTypes.bool,
  jobs: PropTypes.array,
  isLoaded: PropTypes.bool,
  projectName: PropTypes.string,
  allInputParams: PropTypes.array,
  allMetrics: PropTypes.array,
  updateHiddenStatus: PropTypes.func,
  statuses: PropTypes.array,
  updateHiddenUser: PropTypes.func,
  allUsers: PropTypes.array,
  hiddenUsers: PropTypes.array,
  updateNumberFilter: PropTypes.func,
  numberFilters: PropTypes.array,
  updateContainsFilter: PropTypes.func,
  containFilters: PropTypes.array,
  updateBoolFilter: PropTypes.func,
  boolFilters: PropTypes.array,
  boolCheckboxes: PropTypes.array,
  updateDurationFilter: PropTypes.func,
  durationFilters: PropTypes.array,
  updateJobIdFilter: PropTypes.func,
  jobIdFilters: PropTypes.array,
  updateStartTimeFilter: PropTypes.func,
  startTimeFilters: PropTypes.array,
  filters: PropTypes.array,
  selectedRow: PropTypes.string,
  onDataUpdated: PropTypes.func,
  onClickJob: PropTypes.func,
  getJobs: PropTypes.func,
  reload: PropTypes.func,
};

JobTable.defaultProps = {
  isMount: false,
  jobs: [],
  isLoaded: false,
  projectName: '',
  allInputParams: [],
  allMetrics: [],
  updateHiddenStatus: () => {},
  statuses: [],
  updateHiddenUser: () => {},
  allUsers: [],
  hiddenUsers: [],
  updateNumberFilter: () => {},
  numberFilters: [],
  updateContainsFilter: () => {},
  containFilters: [],
  updateBoolFilter: () => {},
  boolFilters: [],
  boolCheckboxes: [],
  updateDurationFilter: () => {},
  durationFilters: [],
  updateJobIdFilter: () => {},
  jobIdFilters: [],
  updateStartTimeFilter: () => {},
  startTimeFilters: [],
  filters: [],
  selectedRow: -1,
  onDataUpdated: () => window.location.reload(),
  onClickJob: job => {},
  getJobs: () => {},
  reload: () => {},
};

export default JobTable;
