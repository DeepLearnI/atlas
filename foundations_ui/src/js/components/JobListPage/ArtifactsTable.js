import React from 'react';
import PropTypes from 'prop-types';
import ArtifactRow from './ArtifactRow';

class ArtifactsTable extends React.Component {
  render() {
    const { job, onClickArtifact } = this.props;
    return (
      <div className="container-artifacts-table">
        <div className="table-artifacts-header">
          <p>Artifact Name</p>
        </div>
        <div className="table-artifacts-header last" />
        {job.artifacts ? job.artifacts.map(artifact => {
          return <ArtifactRow onClickArtifact={onClickArtifact} key={artifact.id} artifact={artifact} />;
        }) : null
        }
      </div>
    );
  }
}

ArtifactsTable.propTypes = {
  job: PropTypes.object,
  location: PropTypes.object,
  onClickArtifact: PropTypes.func,
};

ArtifactsTable.defaultProps = {
  job: {},
  location: { state: {} },
  onClickArtifact: () => {},
};

export default ArtifactsTable;
