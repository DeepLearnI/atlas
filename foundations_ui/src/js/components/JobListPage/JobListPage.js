import React, { Component } from 'react';
import PropTypes from 'prop-types';
import JobTable from './JobTable';
import Toolbar from '../common/Toolbar';
import JobHeader from './JobHeader';
import CommonActions from '../../actions/CommonActions';
import JobActions from '../../actions/JobListActions';

class JobListPage extends Component {
  constructor(props) {
    super(props);
    this.updateHiddenStatus = this.updateHiddenStatus.bind(this);
    this.formatAndSaveParams = this.formatAndSaveParams.bind(this);
    this.saveAPIJobs = this.saveAPIJobs.bind(this);
    this.clearState = this.clearState.bind(this);
    this.saveFilters = this.saveFilters.bind(this);
    this.state = {
      projectName: this.props.projectName,
      project: this.props.project,
      filters: [],
      statuses: [
        { name: 'Completed', hidden: false },
        { name: 'Processing', hidden: false },
        { name: 'Error', hidden: false },
      ],
      jobs: [],
      allInputParams: [],
      isMount: false,
      allMetrics: [],
    };
  }

  async componentDidMount() {
    this.setState({ isMount: true });
    await this.getJobs();
  }

  componentWillUnmount() {
    this.setState({ isMount: false });
  }

  async getJobs() {
    const { projectName } = this.state;
    const apiJobs = await JobActions.getJobs(projectName);
    this.formatAndSaveParams(apiJobs);
  }

  formatAndSaveParams(apiJobs) {
    // use is mount for async as when it returns may have been unmounted
    const { isMount } = this.state;
    if (isMount) {
      if (apiJobs != null) {
        this.saveAPIJobs(apiJobs);
      } else {
        this.clearState();
      }
    }
  }

  clearState() {
    this.setState({
      jobs: [], isLoaded: true, allInputParams: [], allMetrics: [],
    });
  }

  async updateHiddenStatus(hiddenFields) {
    const { statuses, projectName } = this.state;
    const statusNamesArray = statuses.map(status => status.name);
    const formattedColumns = CommonActions.formatColumns(statusNamesArray, hiddenFields);
    const apiFilteredJobs = await JobActions.filterJobs(projectName, formattedColumns);
    this.clearState();
    this.formatAndSaveParams(apiFilteredJobs);
    this.setState({ statuses: formattedColumns });
    this.saveFilters();
    this.forceUpdate();
  }

  saveAPIJobs(apiJobs) {
    const getAllInputParams = JobActions.getAllInputParams(apiJobs.jobs);
    const getAllMetrics = JobActions.getAllMetrics(apiJobs.jobs);
    this.setState({
      jobs: apiJobs.jobs, isLoaded: true, allInputParams: getAllInputParams, allMetrics: getAllMetrics,
    });
  }

  saveFilters() {
    const { filters, statuses } = this.state;
    const newFilters = JobActions.getAllFilters(filters, statuses);
    this.setState({ filters: newFilters });
  }

  render() {
    const {
      projectName, project, filters, statuses, isLoaded, allInputParams, jobs, allMetrics,
    } = this.state;
    return (
      <div className="job-list-container">
        <Toolbar />
        <JobHeader project={project} filters={filters} />
        <JobTable
          projectName={projectName}
          statuses={statuses}
          updateHiddenStatus={this.updateHiddenStatus}
          jobs={jobs}
          isLoaded={isLoaded}
          allInputParams={allInputParams}
          allMetrics={allMetrics}
        />
      </div>
    );
  }
}

JobListPage.propTypes = {
  projectName: PropTypes.string,
  project: PropTypes.object,
  filters: PropTypes.array,
  statuses: PropTypes.array,
  jobs: PropTypes.array,
  allInputParams: PropTypes.array,
  allMetrics: PropTypes.array,
};

JobListPage.defaultProps = {
  projectName: '',
  project: {},
  filters: [],
  statuses: [],
  jobs: [],
  allInputParams: [],
  allMetrics: [],
};

export default JobListPage;
