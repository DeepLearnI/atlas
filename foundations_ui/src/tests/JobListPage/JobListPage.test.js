import React from 'react';
import ReactDOM from 'react-dom';
import JobListPage from '../../js/components/JobListPage/JobListPage';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders Job List Page', () => {
  const wrapper = shallow(<JobListPage/>);
  expect(wrapper).toMatchSnapshot();
});

// Won't actually get anything unless `projectName` is a project name
// in your API
it('Calls Get Job List', async () => {
  <MemoryRouter>
    const wrapper = mount(<JobListPage projectName='myProject'/>); 
    const postState = wrapper.state();
    expect(postState.isLoaded === true);
  </MemoryRouter>
});