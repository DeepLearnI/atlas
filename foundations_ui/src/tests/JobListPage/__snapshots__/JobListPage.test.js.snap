// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Job List Page 1`] = `
<div
  className="job-list-container"
>
  <Toolbar />
  <JobHeader
    bubbleRefs={Array []}
    clearFilters={[Function]}
    filters={Array []}
    hiddenBubbles={Array []}
    isShowingMoreFilters={false}
    numProjects={0}
    project={Object {}}
    removeFilter={[Function]}
  />
  <JobTable
    allInputParams={Array []}
    allMetrics={Array []}
    allUsers={Array []}
    boolCheckboxes={
      Array [
        Object {
          "hidden": false,
          "name": "True",
        },
        Object {
          "hidden": false,
          "name": "False",
        },
      ]
    }
    boolFilters={Array []}
    containFilters={Array []}
    durationFilters={Array []}
    hiddenUsers={Array []}
    isLoaded={false}
    isMount={false}
    jobs={Array []}
    numberFilters={Array []}
    projectName=""
    statuses={
      Array [
        Object {
          "hidden": false,
          "name": "Completed",
        },
        Object {
          "hidden": false,
          "name": "Running",
        },
        Object {
          "hidden": false,
          "name": "Failed",
        },
      ]
    }
    updateBoolFilter={[Function]}
    updateContainsFilter={[Function]}
    updateDurationFilter={[Function]}
    updateHiddenStatus={[Function]}
    updateHiddenUser={[Function]}
    updateNumberFilter={[Function]}
  />
</div>
`;
